File: src/types/mistake.ts
import {Record, Static, String} from 'runtypes';

import Move, {MoveRecord} from './move';

export const MistakeRecord = Record({
  correctDestination: String,
  errantMove: MoveRecord
});

type Mistake = Static<typeof MistakeRecord>;

export default Mistake;

export function createMistake(correctDestination: string, move: Move): Mistake {
  return {
    correctDestination,
    errantMove: move
  };
}

File: java/src/main/java/com/robwilliamson/mailfamiliar/service/predictor/Mistake.java
package com.robwilliamson.mailfamiliar.service.predictor;

import com.robwilliamson.mailfamiliar.entity.Mailbox;
import com.robwilliamson.mailfamiliar.model.Id;
import lombok.Data;

@Data
public class Mistake {
  private final Id<Mailbox> correctDestination;
  private final Move errantMove;
}
